pkg/services/notification/resource_control/task.go

```diff
@@ -63,36 +65,70 @@ func GetTask(taskId string) (*models.Task, error) {
 	return task, nil
 }
 
+func GetTasksByTaskIds(taskIds []string) ([]*models.Task, error) {
+	db := global.GetInstance().GetDB()
+	var tasks []*models.Task
+	err := db.Where("task_id in( ? )", taskIds).Find(&tasks).Error
+	if err != nil {
+		return nil, err
+	}
+	return tasks, nil
+}
+
+func GetTasksByNfId(nfIds []string) ([]*models.Task, error) {
+	var tasks []*models.Task
+	err := nfdb.GetChain(global.GetInstance().GetDB().Where(models.TaskColNfId+" in (?)", nfIds).Find(&tasks)).Error
+	if err != nil {
+		return nil, err
+	}
+	return tasks, nil
+
+}
+
 func DescribeTasks(ctx context.Context, req *pb.DescribeTasksRequest) ([]*models.Task, uint64, error) {
 	req.NotificationId = stringutil.SimplifyStringList(req.NotificationId)
 	req.TaskId = stringutil.SimplifyStringList(req.TaskId)
 	req.TaskAction = stringutil.SimplifyStringList(req.TaskAction)
 	req.ErrorCode = stringutil.SimplifyStringList(req.ErrorCode)
 	req.Status = stringutil.SimplifyStringList(req.Status)
-
-	limit := dbutil.GetLimit(req.Limit)
-	offset := dbutil.GetOffset(req.Offset)
+	offset := pbutil.GetOffsetFromRequest(req)
+	limit := pbutil.GetLimitFromRequest(req)
 
 	var tasks []*models.Task
 	var count uint64
 
-	if err := dbutil.GetChain(globalcfg.GetInstance().GetDB().Table(models.TableTask)).
+	if err := nfdb.GetChain(global.GetInstance().GetDB().Table(models.TableTask)).
 		AddQueryOrderDir(req, models.TaskColCreateTime).
 		BuildFilterConditions(req, models.TableTask).
 		Offset(offset).
 		Limit(limit).
 		Find(&tasks).Error; err != nil {
-		logger.Errorf(ctx, "Describe Tasks failed: %+v", err)
+		logger.Errorf(ctx, "Describe tasks failed, [%+v]", err)
 		return nil, 0, err
 	}
 
-	if err := dbutil.GetChain(globalcfg.GetInstance().GetDB().Table(models.TableTask)).
+	if err := nfdb.GetChain(global.GetInstance().GetDB().Table(models.TableTask)).
 		BuildFilterConditions(req, models.TableTask).
 		Count(&count).Error; err != nil {
-		logger.Errorf(ctx, "Describe Tasks count failed: %+v", err)
+		logger.Errorf(ctx, "Describe Tasks count failed, [%+v]", err)
 		return nil, 0, err
 	}
 
 	return tasks, count, nil
 
 }
+
+func UpdateTasks2Pending(ctx context.Context, taskIds []string) ([]string, error) {
```

Comment:
May be UpdateTasksStatus will be better. 
func UpdateTasksStatus(ctx context.Context, taskIds []string, status string)

Reply:
updated.

Reply:
done